snforge_std/src/cheatcodes/events.cairo

```diff
@@ -121,30 +110,26 @@ fn is_emitted<T, impl TEvent: starknet::Event<T>, impl TDrop: Drop<T>>(
     expected_event.append_keys_and_data(ref expected_keys, ref expected_data);
 
     let mut j = 0;
-    return loop {
-        if j >= emitted_events.len() {
-            break false;
-        }
+    let mut is_emitted = false;
+    while j < emitted_events.len() {
         let (from, event) = emitted_events.at(j);
 
         if from == expected_from && event.keys == @expected_keys && event.data == @expected_data {
             remove_event(ref self, j);
-            break true;
-        }
+            is_emitted = true;
```

Comment:
Does return while .. {} not work?

Reply:
If you meant `break true` inside `while` loop then no, it's not possible